src/_arraykit.c

```diff
@@ -383,7 +383,28 @@ dtype_from_element(PyObject *Py_UNUSED(m), PyObject *arg)
 
     // Integers
     if (PyLong_CheckExact(arg)) {
-        return (PyObject*)PyArray_DescrFromType(NPY_LONG);
+        int overflow;
+        npy_int64 v = PyLong_AsLongLongAndOverflow(arg, &overflow);
+        if (v != -1 && overflow == 0) {
+            // Prioritize fastest case first
+            return (PyObject*)PyArray_DescrFromType(NPY_LONG); // (-2**63, 2**63)
+        }
+        if (v == -1 && PyErr_Occurred()) {
+            return NULL;
+        }
+        if (overflow == -1) {
+            return (PyObject*)PyArray_DescrFromType(NPY_OBJECT); // [-∞, -2**63]
```

Comment:
I'd prefer to keep Unicode symbols our of our source code, just because I know C compiler encodings can be complicated (maybe even locale-specific):

```suggestion
            return (PyObject*)PyArray_DescrFromType(NPY_OBJECT);  // (-inf, -2**63)
```

Reply:
However, I'll replace `∞` with `inf`